src/worker.ts

```diff
@@ -132,32 +98,44 @@ export class Worker {
    * @returns true if the worker is healthy, false otherwise
    */
   async healthCheck() {
-    if (!this.#running) {
+    if (this.#process === null) {
       return false;
     }
-    const res = await fetch(`http://localhost:${this.port}`, {
-      headers: {
-        "X-Health-Check": "true",
-      },
-    });
-    return res.ok;
-  }
-
-  restart() {
-    console.log("[worker]", `${this.name}`, "restarting");
-    if (this.#running) {
-      this.#process?.kill("SIGINT");
-      this.#running = false;
+    const startTime = Date.now();
+    while (Date.now() - startTime < 5000) {
+      try {
+        const res = await fetch(`http://localhost:${this.port}`, {
+          headers: {
+            "X-Health-Check": "true",
+          },
+        });
+        if (res.ok) {
+          return true;
+        }
+      } catch {
+        await new Promise((resolve) => setTimeout(resolve, 20));
+      }
     }
-    return this.start();
+    return false;
   }
 
-  async shutdown() {
+  shutdown() {
+    if (this.#process === null) return;
     console.log("[worker]", `${this.name}`, "shutting down");
-    if (this.#running) {
-      this.#process?.kill("SIGINT");
-    }
-    this.#emitter.removeAllListeners();
+    this.#process.kill("SIGINT");
```

Comment:
If the process crashes early for whatever reason, this will throw. 

![CleanShot 2024-10-28 at 12 41 26@2x](https://github.com/user-attachments/assets/63b4f0a8-7ca5-49c4-91dd-a6d5b9a365d2)